Modules/Quotas/src/Quotas.Domain/Aggregates/Identities/Identity.cs

```diff
@@ -129,4 +129,18 @@ private IEnumerable<Quota> GetAppliedQuotasForMetric(MetricKey metric)
         var appliedQuotas = allQuotasOfMetric.Where(q => q.Weight == highestWeight).ToArray();
         return appliedQuotas;
     }
+
+    public async Task ChangeTierAsync(Tier newTier, MetricCalculatorFactory metricCalculatorFactory, CancellationToken cancellationToken)
+    {
+        _tierQuotas.RemoveAll(_ => true);
+        _metricStatuses.RemoveAll(_ => true);
+
+        TierId = newTier.Id;
+        foreach (var tierQuotaDefinition in newTier.Quotas)
+        {
+            AssignTierQuotaFromDefinition(tierQuotaDefinition);
+        }
+        var metricKeys = newTier.Quotas.Select(q => q.MetricKey);
+        await UpdateMetricStatuses(metricKeys, metricCalculatorFactory, cancellationToken);
```

Comment:
As I said in a comment of the previous review, you should recalculate ALL metric statuses. Because what if the old tier defined a quota that doesn't exist in the new tier? Then the MetricStatus may be exhausted even though it shouldn't be.

Reply:
Didn't realize that you clear the metric statuses. This fixes the issue I described before.

But it introduces another one: you don't take individual Quotas into account. Because imagine an individual quota on Metric M was exhausted before the Tier change. Now you change the Tier, which deletes all metric statuses. Then you only recalculate the ones from the Tier. But if the new Tier doesn't have a Quota on Metric M, it will be unexhausted after the save.

Reply:
Well spotted. What do you think about a method like this?

```c#
private async Task UpdateAllMetricStatuses(MetricCalculatorFactory factory,CancellationToken cancellationToken)
{
    var metricKeys = _tierQuotas.Select(q => q.MetricKey).Union(_individualQuotas.Select(q => q.MetricKey)).Distinct();
    await UpdateMetricStatuses(metricKeys, factory, cancellationToken);
}
```

Reply:
I don't think it's ideal, but we can go with it as an intermediate solution.  You should clear the metric keys in here though.